runtime/integration-tests/src/cases/lp/setup_lp.rs

```diff
@@ -0,0 +1,974 @@
+			}
+			(POOL_B, tranche, names::DAI) if tranche == utils::pool_b_tranche_1_id::<T>() => {
+				names::POOL_B_T_1_DAI
+			}
+			(POOL_B, tranche, names::DAI) if tranche == utils::pool_b_tranche_2_id::<T>() => {
+				names::POOL_B_T_2_DAI
+			}
+			(POOL_C, tranche, names::DAI) if tranche == utils::pool_c_tranche_1_id::<T>() => {
+				names::POOL_C_T_1_DAI
+			}
+
+			(_, _, _) => {
+				unimplemented!("pool, tranche, currency combination does not have a name.")
+			}
+		}
+	};
+
+	for (pool, tranche_id) in [
+		(POOL_A, utils::pool_a_tranche_1_id::<T>()),
+		(POOL_B, utils::pool_b_tranche_1_id::<T>()),
+		(POOL_B, utils::pool_b_tranche_2_id::<T>()),
+		(POOL_C, utils::pool_c_tranche_1_id::<T>()),
+	] {
+		for currency in [names::USDC, names::FRAX, names::DAI] {
+			evm.call(
+				Keyring::Alice,
+				Default::default(),
+				names::POOL_MANAGER,
+				"deployVault",
+				Some(&[
+					Token::Uint(Uint::from(pool)),
+					Token::FixedBytes(FixedBytes::from(tranche_id)),
+					Token::Address(evm.deployed(currency).address()),
+				]),
+			)
+			.unwrap();
+
+			evm.register(
+				lp_name(pool, tranche_id, currency),
+				contracts::LP,
+				Decoder::<H160>::decode(
+					&evm.view(
+						Keyring::Alice,
+						names::POOL_MANAGER,
+						"getVault",
+						Some(&[
+							Token::Uint(Uint::from(pool)),
+							Token::FixedBytes(FixedBytes::from(tranche_id)),
+							Token::Address(evm.deployed(currency).address()),
+						]),
+					)
+					.unwrap()
+					.value,
+				),
+			);
+		}
+	}
+}
+
+/// Initiates tranches on EVM via `DeployTranche` contract and then sends
+/// `add_tranche(pool, tranche_id)` messages for a total of three tranches of
+/// pool A and B.
+pub fn setup_tranches<T: Runtime>(evm: &mut impl EvmEnv<T>) {
+	// AddTranche 1 of A
+	let tranche_id = {
+		let tranche_id = utils::pool_a_tranche_1_id::<T>();
+		assert_ok!(pallet_liquidity_pools::Pallet::<T>::add_tranche(
+			OriginFor::<T>::signed(Keyring::Admin.into()),
+			POOL_A,
+			tranche_id,
+			Domain::EVM(EVM_DOMAIN_CHAIN_ID)
+		));
+
+		utils::process_outbound::<T>(utils::verify_outbound_success::<T>);
+
+		tranche_id
+	};
+	evm.call(
+		Keyring::Alice,
+		Default::default(),
+		names::POOL_MANAGER,
+		"deployTranche",
+		Some(&[
+			Token::Uint(Uint::from(POOL_A)),
+			Token::FixedBytes(FixedBytes::from(tranche_id)),
+		]),
+	)
+	.unwrap();
+	evm.register(
+		names::POOL_A_T_1,
+		contracts::TRANCHE_TOKEN,
+		Decoder::<H160>::decode(
+			&evm.view(
+				Keyring::Alice,
+				names::POOL_MANAGER,
+				"getTranche",
+				Some(&[
+					Token::Uint(POOL_A.into()),
+					Token::FixedBytes(tranche_id.to_vec()),
+				]),
+			)
+			.unwrap()
+			.value,
+		),
+	);
+	// FIXME: Fails
+	// evm.register(
+	// 	names::RM_POOL_A_T_1,
+	// 	contracts::RESTRICTION_MANAGER,
+	// 	Decoder::<H160>::decode(
+	// 		&evm.view(
+	// 			Keyring::Alice,
+	// 			names::POOL_A_T_1,
+	// 			"restrictionManager",
+	// 			None,
+	// 		)
+	// 		.unwrap()
+	// 		.value,
+	// 	),
+	// );
```

Comment:
TODO

Reply:
Should I include fixing these as part of https://github.com/centrifuge/centrifuge-chain/pull/1933?

Reply:
What you think is better